### PR TITLE
chore(repo): Replace clerk.dev with clerk.com

### PR DESCRIPTION
**Pull request by dimkl**
## Description

Since we have completely migrated to clerk.com, this PR replaces all references of the previous domain (`clerk.dev`) to `clerk.com`.
## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [x] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [x] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [x] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [x] `@clerk/clerk-expo`
- [x] `@clerk/backend`
- [x] `@clerk/clerk-sdk-node`
- [x] `@clerk/shared`
- [x] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [x] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: packages/clerk-js/src/utils/email.ts
Comment (panteliselef) [inline packages/clerk-js/src/utils/email.ts:7]: @dimkl our website still mentions `support@clerk.dev`, let's make sure we support the `.com` and if we do let's create a ticket in order to update it for the marketing as well.

image

Comment (dimkl) [reply packages/clerk-js/src/utils/email.ts:7]: Based on the implementation and usage of `buildEmailAddress()` we use the frontendApi to generate the email domain. This change probably won't affect the support email address.
I will test it more before merging this.

File: packages/clerk-js/src/ui/customizables/makeResponsive.tsx
Comment (anagstef) [inline packages/clerk-js/src/ui/customizables/makeResponsive.tsx:36]: ❓ Maybe keep this one for backwards compatibility? If not, then `.com` entry already exists.

**Review by agis — APPROVED** [review]
LGTM (only one note regarding backwards compatibility)! 🚀

File: packages/clerk-js/src/ui/customizables/makeResponsive.tsx
Comment (dimkl) [reply packages/clerk-js/src/ui/customizables/makeResponsive.tsx]: let us proceed with keeping only the `.com`. Monitor the logs and then drop the `img.clerk.dev`  domain after some time (eg 6 months) if most of the traffic is migrated. wdyt?

File: packages/backend/src/constants.ts
Comment (dimkl) [reply packages/backend/src/constants.ts:1]: > strict firewalls configured to only allow outgoing traffic to api.clerk.dev and not api.clerk.com

@agis Could you check the firewall rules if we support the clerk.com for all of our workers services (eg img.clerk.com)?  If so, I think we could proceed with this PR.

> test suites asserting things like expect(request).to('api.clerk.dev')

Could you elaborate more on this? In the javascript repo, I believe that all the tests are updated to use `clerk.com`

Comment (agis) [reply packages/backend/src/constants.ts:1]: I was referring to customers' test suites and WAF, not our own (our own WAF is fine, since `clerk.com` is already used by a handful of customers already). I was just trying to envision any cases where changing this constant's value would break customers.

Feel free to disregard my comment if you believe this is highly unlikely.

**Review by anagstef — APPROVED** [review]
LGTM

File: packages/backend/src/constants.ts
Comment (dimkl) [reply packages/backend/src/constants.ts:1]: Based on the above comments i think we should introduce the change of `clerk.dev -> `clerk.com`) as part of the next major version and add references in the migration guide and changelog about changing the clerk domains to avoid causing issues in customers depending the requests to our systems (eg mocking the requests).

**Comment by dimkl** [comment]
Blocked by https://github.com/clerkinc/javascript/pull/1837

**Review by nikosdouvlis — APPROVED** [review]
Nice cleanup @dimkl

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**
